### PR TITLE
Moves grown food to newfood

### PR DESCRIPTION
**Pull request by ghost**
:cl: Qustinnus
 code: Moves grown food to newfood
 code: Gives trash element support for callbacks for item creation override
/:cl:

thank fucking god. After this pr (and the other open newfood prs) its literaly just the left-over references to the old typepath, and then we are done for phase 1.

dont mind files changed; I used my miscfood as a main branch due to me needing some of the commits on it for this pr.

### CODE REVIEW

**Comment by brotherbeyondd** [comment]
Would this indirectly nerf plantbags since all newfoods are normal sized rather than tiny?

**Comment by Qustinnus** [comment]
I'll update the plantbag so balance doesn't change. Thanks for bringing that up!

**Comment by Qustinnus** [comment]
I also need to update w_class for these in general, a melon != an apple

File: code/modules/unit_tests/serving_tray.dm [hunk 1]
```diff
@@ -5,7 +5,7 @@
 	var/mob/living/carbon/human/human = allocate(/mob/living/carbon/human)
 	var/obj/structure/table/the_table = allocate(/obj/structure/table)
 	var/obj/item/storage/bag/tray/test_tray = allocate(/obj/item/storage/bag/tray)
-	var/obj/item/reagent_containers/food/banana = allocate(/obj/item/reagent_containers/food/snacks/grown/banana)
+	var/obj/item/reagent_containers/food/banana = allocate(/obj/item/food/grown/banana)
```
Comment (Jared-Fogle) [inline code/modules/unit_tests/serving_tray.dm:8]:
```suggestion
	var/obj/item/food/banana = allocate(/obj/item/food/grown/banana)
```

Comment (Jared-Fogle) [inline code/modules/unit_tests/serving_tray.dm:8]: Change this to another food that is an `/obj/item/reagent_containers/food/snacks`--the point of this test is to ensure both old food and new food work on serving trays, and we already have a new food underneath (`/obj/item/food/the_bread`).

**Pull request merged**